### PR TITLE
Split package for poppler-qt

### PR DESCRIPTION
**Pull request by izahn**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

closes https://github.com/conda-forge/poppler-feedstock/issues/113

I've really struggled to get a split package working, appreciate any help!

### CODE REVIEW

**Comment by pkgw** [comment]
I think we might want to emulate the style of the `gobject-introspection` feedstock, which just has one build script where the resulting files are split. I'm playing around with trying that approach but haven't gotten it working just yet.

**Comment by pkgw** [comment]
See #115. I was at least able to get it to build on linux-64, although it will probably need work for other OSs and cross builds.

**Comment by izahn** [comment]
This is wonderful, thank you!

File: recipe/meta.yaml [hunk 1]
```diff
@@ -64,16 +74,140 @@ requirements:
...
+      - {{ prefix }}share/man/man1/pdfimages.1
+      - {{ prefix }}share/man/man1/pdfinfo.1
+      - {{ prefix }}share/man/man1/pdfseparate.1
+      - {{ prefix }}share/man/man1/pdftocairo.1
+      - {{ prefix }}share/man/man1/pdftohtml.1
+      - {{ prefix }}share/man/man1/pdftoppm.1
+      - {{ prefix }}share/man/man1/pdftops.1
+      - {{ prefix }}share/man/man1/pdftotext.1
+      - {{ prefix }}share/man/man1/pdfunite.1
+      - {{ prefix }}Library/bin/pdfattach.exe
```
Comment (pkgw) [inline recipe/meta.yaml:122]: Note that at the top of this file, `{{ prefix }}` is defined to include `Library/` on Windows. So you should be able to get rid of most of these; I think there is a variable `EXEEXT` or something that expands to `.exe` on Windows, for the files in `bin`?

Comment (izahn) [reply recipe/meta.yaml]: Oh I did miss that, thanks! I did see some error earlier that prompted me to add the `Library` prefix by hand, but I'll go back and clean it up.

**Comment by izahn** [comment]
I think this is starting to look pretty good except that the windows test fails. I can't understand what the error is trying to tell me. If you have time to take a look at the windows build log that would be awesome!

**Comment by izahn** [comment]
Thanks to @pkgw this is ready for review!

**Comment by xhochy** [comment]
From my learnings in the past, I would recommend to not use the files approach but rather use `conda-build`'s deduplication as we use in the `glib` feedstock. This is less brittle as we might otherwise omit files in the  `poppler` output that are added with newer versions. See https://github.com/conda-forge/glib-feedstock/blob/1bd2a53a047ca7cc52632b67ddf7d62fcc0e339f/recipe/install.sh#L13

I would add the following pseudo code to install.sh:

```
make install
if pkg_name == poppler; then
  remove poppler qt files
fi
```

That should be sufficient and get us rid of the explicit file listing for `poppler`.

**Comment by izahn** [comment]
Thanks @xhochy that makes sense, I'll change to script based outputs.

One other doubt; should I set the version here to `21.03` since that is the current global pin in https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/eae23e911d84334ca1c233004b28841311fd091a/recipe/conda_build_config.yaml#L626 ? My goal is to be able to use `poppler-qt` to build other packages, e.g., in `staged-recipes` submissions.

**Comment by izahn** [comment]
I switched to install scripts in https://github.com/conda-forge/poppler-feedstock/pull/114/commits/b1c121fcbeab86760df137ee16b9af5dd7ce7834 though I'm not entirely convinced that approach is better. It seems to me that either way we end up making  lists of files to include (or remove) and things will break it they change. I don't have a preference, feel free to keep or revert https://github.com/conda-forge/poppler-feedstock/pull/114/commits/b1c121fcbeab86760df137ee16b9af5dd7ce7834 as you prefer.

**Comment by izahn** [comment]
Well, it turns out that my script-based approach failed on OSX. The problem seems to be that `ninja install` wants to re-run the build instead of just copying things as I expected it to. The build log is at https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=378373&view=logs&jobId=58ac6aab-c4bc-5de2-2894-98e408cc8ec9&j=58ac6aab-c4bc-5de2-2894-98e408cc8ec9. My preference would be to revert https://github.com/conda-forge/poppler-feedstock/commit/b1c121fcbeab86760df137ee16b9af5dd7ce7834 but I'm open to other suggestions.

**Comment by izahn** [comment]
@xhochy I tried to get a script-based split package working, but it seemed even more fragile than the file-based method. I've reverted back to file-based here because that is the best I can currently get working.

After spending this time I'm personally not sure this is even worth it -- maybe easier to just have a totally separate `poppler-qt`, or just include it in a monolithic `poppler` package. I'm saying, split packages are kind of a pain...

**Comment by pkgw** [comment]
@izahn Thanks for your persistence! This is close to working so I think it's worth pushing on through. I may try implementing @xhochy's suggestion and see if I can solve the problems you were having. One of these days we'll have a perfect package manager ...

**Comment by izahn** [comment]
Thanks @pkgw and @xhochy appreciate all your help! Let me know if there is anything more I can do here.

**Comment by izahn** [comment]
It looks good to me!

**Comment by izahn** [comment]
Closing in favor of https://github.com/conda-forge/poppler-feedstock/pull/115

**Pull request closed without merging**